### PR TITLE
Re-enable Vault tests

### LINKED ISSUE #11879: quarkus-integration-test-vault needs to be fixed
This module contains tests that make use of `io.quarkus.test.QuarkusUnitTest` from `io.quarkus:quarkus-junit5-internal` (such tests should reside in the `quarkus-vault-deployment` module). As a result a weird class loading behavior is observed.

NOTE: The problem was identified while investigating a failure in https://github.com/quarkusio/quarkus/pull/11848. For some reason the `VaultHealthCheck` class was considered an application class and included in the bean archive but `org.eclipse.microprofile.health.HealthCheck` that is referenced by this class was not visible when running the test (`io.quarkus:quarkus-smallrye-health` is an optional dependency of `io.quarkus:quarkus-vault`).

### LINKED ISSUE #12242: Vault health check not disabled as documented, and explicit disable does not work
**Describe the bug**

Vault healthcheck property does not work as documented.

**Expected behavior**

Default behavior, as documented, is that smallrye healthchecks should _not_ include vault (Default is "false" for property quarkus.vault.health.enabled).

Additionally, explicitly setting `quarkus.vault.health.enabled=false` should disable Vault healthchecks in smallrye health.

**Actual behavior**

Default behavior appears to be "true", to enable healthchecks, and explicitly setting the documented property to "false" does not disable smallrye health incorporation of Vault health.

The incorrect default behavior happens in dev-mode (quarkus:dev) and when starting via jar.

Explicitly setting the property `quarkus.vault.health.enabled=false` does not work in dev-mode (quarkus:dev) or when starting via jar, after building with that property set.

**To Reproduce**

Reproducer: https://github.com/jeffhubLR/quarkus-1.8-vault-health
Project generated from code.quarkus.io, enabling smallrye health and vault, and setting `quarkus.vault.url` to match the provided docker-compose file.

A second commit in that repo attempted to explicitly disable healthchecks via `quarkus.vault.health.enabled=false`

A third commit in that repo adds README steps to also test against -runner.jar

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.vault.health.enabled=false
```

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin USMACEA094JHUBB 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
 - Output of `java -version`: openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)
 - GraalVM version (if different from Java): N/A
 - Quarkus version or git rev: 1.8.0.Final and 1.8.1.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/Cellar/maven/3.6.3_1/libexec
Java version: 11.0.8, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.0.8.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.5", arch: "x86_64", family: "mac"

**Additional context**
It appears Vault healthcheck was originally added as a response to this issue: https://github.com/quarkusio/quarkus/issues/7007
So related code should be linked there.

Documentation link for the expected behavior and property in question: https://quarkus.io/guides/vault#quarkus-vault_quarkus.vault.health.enabled

### PR DESCRIPTION
**Pull request by vsevel**
Fixes #12242 and #11879
Alternative implementation to https://github.com/quarkusio/quarkus/pull/12293

### CODE REVIEW

File: extensions/vault/model/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        <artifactId>quarkus-vault-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-vault-model</artifactId>
+    <name>Quarkus - Vault - Model</name>
+    <description>Store your credentials securely in HashiCorp Vault</description>
```
Comment (jmartisk) [inline extensions/vault/model/pom.xml:15]: This description should probably explain why this module is separate from the rest

Comment (jmartisk) [reply extensions/vault/model/pom.xml:15]: Or rather that these are the model classes; 
and the explanation why they are separate could be... in a comment maybe?

Comment (vsevel) [reply extensions/vault/model/pom.xml:15]: yes. I focused on the technical part at this point. it has just been copy/pasted. there a few details to polish. thanks.

**Comment by vsevel** [comment]
I have added a comment to explain the rationale behind this module.
the vault model module gets indexed through an `IndexDependencyBuildItem`. is there a better way to do it?
https://github.com/vsevel/quarkus/blob/issue_vault_11879_bis/extensions/vault/deployment/src/main/java/io/quarkus/vault/VaultProcessor.java#L32-L35

**Comment by jmartisk** [comment]
I think that's fine. We could also put a `beans.xml` there but I'd prefer avoiding that

**Review by jmartisk — APPROVED** [review]

**Review by geoand — APPROVED** [review]
:+1:

**Comment by vsevel** [comment]
might fix https://github.com/quarkusio/quarkus/issues/11753 as well

**Comment by vsevel** [comment]
kudo for @jmartisk 's analysis https://github.com/quarkusio/quarkus/issues/12242#issuecomment-697227480

**Pull request merged**